eagleproject/core/blockchain.py

```diff
@@ -472,6 +520,70 @@ def ensure_staking_snapshot(block_number):
     )
 
 
+def ensure_oracle_snapshot(block_number):
+    """ Take oracle snapshots """
+    existing_snaps = OracleSnapshot.objects.filter(block_number=block_number)
+    if len(existing_snaps) > 0:
+        return existing_snaps
+
+    snaps = []
+
+    # USD price of ETH
+    usd_eth_price = chainlink_ethUsdPrice()
+    if usd_eth_price:
+        snaps.append(
+            OracleSnapshot.objects.create(
+                block_number=block_number,
+                oracle=CHAINLINK_ORACLE,
+                ticker_left="ETH",
+                ticker_right="USD",
+                price=usd_eth_price
+            )
+        )
+
+    # Get oracle prices for all OUSD minting assets
+    for ticker in ASSET_TICKERS:
+        # ETH price
+        eth_price = chainlink_tokEthPrice(ticker)
+        if eth_price:
+            snaps.append(
+                OracleSnapshot.objects.create(
+                    block_number=block_number,
+                    oracle=CHAINLINK_ORACLE,
+                    ticker_left=ticker,
+                    ticker_right="ETH",
+                    price=eth_price
+                )
+            )
+
+        # USD price
+        # Doesn't currently work?  reverts: "Price is not direct to usd"
```

Comment:
Yeah, chainlink does not have direct to USD oracles for USDT and USDC. If they did, we wouldn't need to monkey around with the conversions from ETH.